cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -304,10 +304,9 @@ public List<RunCountResult> getProgramTotalRunCounts(List<ProgramReference> prog
    */
   public RunRecordDetail getRunRecordMeta(ProgramRunId programRunId) throws Exception {
     accessEnforcer.enforce(programRunId, authenticationContext.getPrincipal(), StandardPermission.GET);
-
-    ProgramSpecification programSpec = getProgramSpecificationWithoutAuthz(programRunId.getParent());
-    if (programSpec == null) {
-      throw new NotFoundException(programRunId.getParent());
+    ApplicationMeta appMeta = store.getLatest(programRunId.getParent().getAppReference());
+    if (appMeta == null) {
```

Comment:
do we need this check at all? seems like null check on RunRecordDetail below should be sufficient already?

Reply:
Yeah I think we need to define the behavior here. If the run records exists but the application is being deleted by users: https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java#L603-L604 (the process could take some time and it's an edge case), should we return 404 for appnotfound or still return the runrecord?